### ISSUE TITLE
auto-indentation for code blocks

### ISSUE THREAD

**Issue by denfromufa** [scopatz/xonsh#600]

**Comment by scopatz** [comment]
I believe that we already have [this for the readline shell](https://github.com/scopatz/xonsh/blob/master/xonsh/readline_shell.py#L133).  I believe you use the prompt-toolkit shell, so is this a request for that? Please provide more information on the expected behaviour so that we can be sure. Thanks!

**Comment by denfromufa** [comment]
I was using prompt_toolkit but with pyreadline the situation is much worse:

```
-DENIS ~ # for i in range(5):
pre_input_hook failed
Readline internal error
Traceback (most recent call last):
  File "c:\python\python27\envs\py34\lib\site-packages\pyreadline\modes\basemode
.py", line 124, in readline_setup
    self.pre_input_hook()
  File "c:\python\python27\envs\py34\lib\site-packages\xonsh\readline_shell.py",
 line 82, in inserter
    readline.redisplay()
AttributeError: 'module' object has no attribute 'redisplay'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\python\python27\envs\py34\lib\site-packages\pyreadline\console\consol
e.py", line 768, in hook_wrapper_23
    res = ensure_str(readline_hook(prompt))
  File "c:\python\python27\envs\py34\lib\site-packages\pyreadline\rlmain.py", li
ne 569, in readline
    self.readline_setup(prompt)
  File "c:\python\python27\envs\py34\lib\site-packages\pyreadline\rlmain.py", li
ne 564, in readline_setup
    BaseReadline.readline_setup(self, prompt)
  File "c:\python\python27\envs\py34\lib\site-packages\pyreadline\rlmain.py", li
ne 253, in readline_setup
    return self.mode.readline_setup(prompt)
  File "c:\python\python27\envs\py34\lib\site-packages\pyreadline\modes\basemode
.py", line 127, in readline_setup
    traceback.print_exc()
NameError: name 'traceback' is not defined
  File "<string>", line None
SyntaxError: None: no further code
-DENIS ~ # import sys
-DENIS ~ # 1+1
2
-DENIS ~ #
```

**Comment by scopatz** [comment]
#601 should hopefully address this.

**Comment by denfromufa** [comment]
@scopatz #601 does not address this nested case on the second level:

```
-DENIS ~ # 
                              for i in range(5):
                                    if i % 2 == 0:
                                        print(i)
                                    else:
                                        print(i**2)

0
1
2
9
4
```

**Comment by scopatz** [comment]
#601 now does multi-level indents.

**Comment by gforsyth** [comment]
Hey @denfromufa , I believe this was resolved by #601 and is merged into master -- please reopen if you're still having issues with the auto-indent.

**Comment by denfromufa** [comment]
I checked and it is working! Thank you!
